bin/run-circleci-tests

```diff
@@ -4,6 +4,18 @@ set -e
 
 make install_system_dependencies
 
+# Ensure Redis is up and running before starting to run any tests:
+# In the past, individual tests from our unit test suite
+# would sometimes fail with a "ConnectionRefusedError"
+# caused by Redis not starting for the container running the tests.
+# The solution that we use below comes from the following support thread:
+# https://discuss.circleci.com/t/redis-server-problems-unable-to-connect-localhost-6379/4040
+until bash -c "sudo service redis-server status"; do
```

Comment:
Precisely something like this, just make sure that run succeeds if we wait 15 minutes for Redis (AFAIK we are pretty close to maximal build time in CircleCI for integration tests).